galley/pkg/config/analysis/analyzers/testdata/deprecation.yaml

```diff
@@ -0,0 +1,81 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: route-sidecars
+spec:
+  hosts:
+    - eu.bookinfo.com
+  gateways:
+    - mesh
+  http:
+    - match:
+        - port: 80
+      route:
+        - destination:
+            host: istio-egressgateway.istio-system.svc.cluster.local
+            port:
+              number: 443
+          weight: 100
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: istio-egressgateway
+spec:
+  selector:
+    istio: egressgateway
+  servers:
+    - port:
+        number: 443
+        name: https
+        protocol: HTTPS
+      hosts:
+        - "*"
+      tls:
+        mode: MUTUAL
+        serverCertificate: /etc/certs/cert-chain.pem
+        privateKey: /etc/certs/key.pem
+        caCertificates: /etc/certs/root-cert.pem
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: route-egressgateway
+spec:
+  hosts:
+    - eu.bookinfo.com
+  gateways:
+    - istio-system/istio-egressgateway
+  http:
+    - match:
+        - authority:
+            exact: eu.bookinfo.com
+          port: 443
+      route:
+        - destination:
+            host: eu.bookinfo.com
+            port:
+              number: 80
+          weight: 100
+      appendHeaders:
+        handled-by-egress-gateway: "true"
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: service-entry-bookinfo
+spec:
+  hosts:
+    - eu.bookinfo.com
+  ports:
+    - number: 80
+      name: http
+      protocol: HTTP
+  resolution: DNS
+  endpoints:
+    # Rather than relying on an external host that might become unreachable (causing test failures)
```

Comment:
I left a lot of junk from testing in there because I test my new validators interactively as I develop them, with `make istioctl && ~/go/out/darwin_amd64/release/istioctl x analyze <file> [--use-kube]`.  The analysis framework only checks a specific validator but seeing output unrelated to the validator under test made it hard to see my own changes.

If you want me to clean up the test yaml I will.  Do you have any tricks for testing a single validator other than `grep`?

Reply:
Right now? Comment out the testcases you don't want. :) 

That's a pretty crappy option though. I want to add a CLI option to select analyzers by name, see https://github.com/istio/istio/issues/17443.